core/src/test/java/org/jdbi/v3/core/argument/TestArgumentFactory.java

```diff
@@ -100,7 +100,7 @@ public void testOnPreparedBatch() throws Exception
         public Optional<Argument> build(Type expectedType, Object value, ConfigRegistry config) {
             if (expectedType == Name.class || value instanceof Name) {
                 Name nameValue = (Name) value;
-                return config.findArgumentFor(String.class, nameValue.getFullName());
+                return config.get(Arguments.class).findFor(String.class, nameValue.getFullName(), config);
```

Comment:
It does feel weird to pass the config to something you just got _from_ the config. I've debated whether the config registry  ought to optionally inject itself into config types.